app/src/main/java/de/westnordost/streetcomplete/quests/max_speed/AddMaxSpeedForm.kt

```diff
@@ -87,80 +86,66 @@ class AddMaxSpeedForm : AbstractQuestFormAnswerFragment<MaxSpeedAnswer>() {
         speedInput?.addTextChangedListener(TextChangedWatcher { checkIsFormComplete() })
 
         speedUnitSelect = rightSideContainer.findViewById(R.id.speedUnitSelect)
-        val measurementUnits = countryInfo.measurementSystem
-        speedUnitSelect?.visibility = if (measurementUnits.size == 1) View.GONE else View.VISIBLE
-        speedUnitSelect?.adapter = ArrayAdapter(context!!, R.layout.spinner_item_centered, getSpinnerItems(measurementUnits))
+        speedUnitSelect?.visibility = if (speedUnits.size == 1) View.GONE else View.VISIBLE
+        speedUnitSelect?.adapter = ArrayAdapter(context!!, R.layout.spinner_item_centered, speedUnits)
         speedUnitSelect?.setSelection(0)
 
         checkIsFormComplete()
     }
 
-    private fun getSpinnerItems(units: List<String>) = units.mapNotNull {
-        when(it) {
-            "metric" -> "km/h"
-            "imperial" -> "mph"
-            else -> null
-        }
-    }
-
     private fun getSpeedType(@IdRes checkedId: Int) = when (checkedId) {
-        R.id.sign -> SpeedType.SIGN
-        R.id.zone -> SpeedType.ZONE
-        R.id.no_sign -> SpeedType.NO_SIGN
+        R.id.sign ->    SIGN
+        R.id.zone ->    ZONE
+        R.id.no_sign -> NO_SIGN
         else -> null
     }
 
     private val SpeedType.layoutResId get() = when (this) {
-        AddMaxSpeedForm.SpeedType.SIGN -> R.layout.quest_maxspeed_sign
-        AddMaxSpeedForm.SpeedType.ZONE -> R.layout.quest_maxspeed_zone_sign
-        AddMaxSpeedForm.SpeedType.ADVISORY -> R.layout.quest_maxspeed_advisory
+        SIGN ->     R.layout.quest_maxspeed_sign
+        ZONE ->     R.layout.quest_maxspeed_zone_sign
+        ADVISORY -> R.layout.quest_maxspeed_advisory
         else -> null
     }
 
     private fun userSelectedUnusualSpeed(): Boolean {
-        val speed = speed.toInt()
-        val speedUnit = speedUnitSelect?.selectedItem as String?
-        val speedInKmh = if (speedUnit == "mph") mphToKmh(speed.toDouble()) else speed.toDouble()
-        return speedInKmh > 140 || speed > 20 && speed % 5 != 0
+        val kmh = getSpeedFromInput()?.toKmh() ?: return false
+        return kmh > 140 || kmh > 20 && kmh % 5 != 0
```

Comment:
@westnordost This throws an "implausible" warning for 20 mph in the UK, and would do the same for 30. The % 5 bit needs to be done on the raw speed value rather than the converted kmh value.

I'm not quite sure how I didn't hit this before, or is this only triggered in some cases (this was a zone not a sign).

Reply:
https://github.com/westnordost/StreetComplete/pull/1632

Reply:
Oh, thank you for reporting it and thank you even more for offering a fix in a PR!